### PR TITLE
CORDOPLUG-222

### PR DESCRIPTION
**Pull request by steinwelberg**

### CODE REVIEW

File: docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# OTP mobile authentication Handler
+
+<!-- toc -->
+
+## Introduction
+
+When a user is enrolled for OTP mobile authentication, they are able to receive and respond to mobile authentication requests. After initiating an OTP request, the Onegini Cordova plugin returns a `MobileAuthenticationHandler` object, which can be used to register callbacks which will be called when certain OTP mobile authentication steps are required.
```
Comment (robertszuba) [inline docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md]: What does it mean that "a user is enrolled for OTP mobile authentication"? We have enrollment for mobile authentication (in general) and probably some settings that have to be enabled in TS config in order to perform OTP authentication. But do we have something that can be called "OTP enrollment"?

File: docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+**Example:** When an OTP mobile authentication request is initiated which does not require additional verification (such as PIN or Fingerprint authentication), the `onConfirmationRequest` callback will be called.
```
Comment (robertszuba) [inline docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md]: This sentence indicates that some OTP requests require additional verification (PIN or fingerprint). AFAIK the user have to be already authenticated, but except that, there's no additional verification other than confirmation request.

File: docs/public/reference/mobileAuthentication/otp/handleRequest.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+## `onegini.mobileAuth.otp.handleRequest`
+
+This method is used to trigger mobile authentication with OTP. 
+
+| Parameter | Description |
+| --- | --- |
+| `otp` | The base64 encoded OTP
+
+Currently, the available method strings are `"confirmation"` (push), `"pin"` (push with PIN), and `"fingerprint"` (push with fingerprint).
```
Comment (robertszuba) [inline docs/public/reference/mobileAuthentication/otp/handleRequest.md]: OTP with additional authentication??

File: docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md (hunk 1 above)
Comment (steinwelberg) [inline docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md]: You're right, that should be "enrolled for mobile authentication".

File: docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md (hunk 2 above)
Comment (steinwelberg) [inline docs/public/reference/mobileAuthentication/otp/OtpMobileAuthHandler.md]: Please remove that part of the sentence, we haven't implemented that currently.

File: docs/public/reference/mobileAuthentication/otp/handleRequest.md (hunk 3 above)
Comment (steinwelberg) [inline docs/public/reference/mobileAuthentication/otp/handleRequest.md]: Please remove the parts that we currrently don't support, it's a copy/paste error ;)

File: src/ios/classes/OGCDVOtpMobileAuthRequestClient.m [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+{
+    [self.commandDelegate runInBackground:^{
+        NSDictionary *options = command.arguments[0];
+        NSString *otp = options[OGCDVPluginKeyOtp];
+
+        ONGUserProfile *user = [[ONGUserClient sharedInstance] authenticatedUserProfile];
+        if (user == nil) {
+            [self sendErrorResultForCallbackId:command.callbackId withErrorCode:OGCDVPluginErrCodeNoUserAuthenticated andMessage:OGCDVPluginErrDescriptionNoUserAuthenticated];
+            return;
+        }
```
Comment (stnslw) [inline src/ios/classes/OGCDVOtpMobileAuthRequestClient.m]: is this check necessary?  handleOTPMobileAuthRequest:delegate: method validates if the user is authenticated or not and returns the error on the delegate

File: src/ios/classes/OGCDVOtpMobileAuthRequestClient.m [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+- (void)userClient:(ONGUserClient *)userClient didReceiveConfirmationChallenge:(void (^)(BOOL confirmRequest))confirmation forRequest:(ONGMobileAuthRequest *)request
+{
+    self.mobileAuthRequest = request;
+    self.confirmationChallengeConfirmationBlock = confirmation;
+
+    NSMutableDictionary *message = [[NSMutableDictionary alloc] init];
+    message[OGCDVPluginKeyType] = request.type;
+    message[OGCDVPluginKeyMessage] = request.message;
+    message[OGCDVPluginKeyProfileId] = request.userProfile.profileId;
+    message[OGCDVPluginKeyEvent] = OGCDVPluginEventConfirmationRequest;
```
Comment (stnslw) [inline src/ios/classes/OGCDVOtpMobileAuthRequestClient.m]: Shouldn't we also pass transaction_id within the message dictionary?

File: src/ios/classes/OGCDVOtpMobileAuthRequestClient.m (hunk 4 above)
Comment (steinwelberg) [inline src/ios/classes/OGCDVOtpMobileAuthRequestClient.m]: No, then it's not :) We'll clean it up in a later release.

File: src/ios/classes/OGCDVOtpMobileAuthRequestClient.m (hunk 5 above)
Comment (steinwelberg) [inline src/ios/classes/OGCDVOtpMobileAuthRequestClient.m]: If it is added to the push message handling then we should add it otherwise leave it as it is.

**Pull request merged**